### PR TITLE
OPENEUROPA-1649: eulogin default behavior on first login

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

Block users if administrator approval is required. Allow users to be created in Drupal if appropriate permissions are set.

### Change log

- Added:
- Changed: Block users if administrator approval is required.
- Deprecated:
- Removed: Restriction on user creation
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: src/Event/EuLoginEventSubscriber.php [hunk 1]
```diff
@@ -19,6 +21,7 @@
  * the required modifications to work with EU Login.
  */
 class EuLoginEventSubscriber implements EventSubscriberInterface {
+  use StringTranslationTrait;
```
Comment (upchuk) [inline src/Event/EuLoginEventSubscriber.php]: Empty line needed.

File: src/Event/EuLoginEventSubscriber.php [hunk 2]
```diff
@@ -68,6 +81,14 @@ public static function getSubscribedEvents(): array {
    */
   public function generateEmail(CasPreRegisterEvent $event): void {
     $attributes = $event->getCasPropertyBag()->getAttributes();
+    $user_settings = $this->configFactory->get('user.settings');
```
Comment (upchuk) [inline src/Event/EuLoginEventSubscriber.php]: This method has gone way past its respos. So let's either rename it to be more generic or break it up into multiple ones.

File: tests/features/ecas-login.feature [hunk 3]
```diff
@@ -43,17 +44,13 @@ Feature: Login through OE Authentication
     And I should see the link "Log in"
 
   Scenario: A blocked user should not be able to log in
```
Comment (upchuk) [inline tests/features/ecas-login.feature]: The scenario name no longer reflects what we test. Now we test that by default, any new users are created as blocked.

File: tests/features/ecas-login.feature [hunk 4]
```diff
@@ -43,17 +44,13 @@ Feature: Login through OE Authentication
...
 
     # When I try to log in again I will be denied access.
-    When I click "Log in"
+    When I am on the homepage
+    And I click "Log in"
     And I fill in "Username or e-mail address" with "Lisbeth.SALANDER@ext.ec.europa.eu"
     And I fill in "Password" with "dragon_tattoo"
     And I press the "Login!" button
     Then I should see "There was a problem logging in, please contact a site administrator."
+    And I should see "Thank you for applying for an account. Your account is currently pending approval by the site administrator."
```
Comment (upchuk) [inline tests/features/ecas-login.feature]: We are not actually asserting the user is blocked, only that s/he sees the message. Let's add a check the ensure the user is actually blocked.

Comment (dxvargas) [reply tests/features/ecas-login.feature]: Added a step "And I should see the link "Log in"" that ensures user is actually blocked, marking as resolved.

File: tests/features/ecas-login.feature (hunk 3 above)
Comment (dxvargas) [reply tests/features/ecas-login.feature]: Updated to "A site that requires administration validation on users should block them by default", marking the conversation as resolved.

File: src/Event/EuLoginEventSubscriber.php (hunk 2 above)
Comment (dxvargas) [reply src/Event/EuLoginEventSubscriber.php]: Method was renamed to "processUserProperties", conversation can be resolved.

File: src/Event/EuLoginEventSubscriber.php (hunk 1 above)
Comment (dxvargas) [reply src/Event/EuLoginEventSubscriber.php]: Done.

**Pull request merged**